internal/nodes/avi_model_l4_translator.go

```diff
@@ -100,21 +102,20 @@ func (o *AviObjectGraph) ConstructAviL4VsNode(svcObj *corev1.Service, key string
 		// Default case
 		avi_vs_meta.ApplicationProfile = utils.DEFAULT_L4_APP_PROFILE
 	}
-	if !isTCP {
-		if isSCTP {
-			avi_vs_meta.NetworkProfile = utils.SYSTEM_SCTP_PROXY
-		} else {
-			avi_vs_meta.NetworkProfile = utils.SYSTEM_UDP_FAST_PATH
-		}
 
-	} else {
+	if isSCTP && !isTCP && !isUDP {
+		avi_vs_meta.NetworkProfile = utils.SYSTEM_SCTP_PROXY
+	} else if isTCP && !isUDP && !isSCTP {
 		license := lib.AKOControlConfig().GetLicenseType()
-
 		if license == lib.LicenseTypeEnterprise {
 			avi_vs_meta.NetworkProfile = utils.DEFAULT_TCP_NW_PROFILE
 		} else {
 			avi_vs_meta.NetworkProfile = utils.TCP_NW_FAST_PATH
 		}
+	} else if isUDP && !isTCP && !isSCTP {
+		avi_vs_meta.NetworkProfile = utils.SYSTEM_UDP_FAST_PATH
+	} else {
+		avi_vs_meta.NetworkProfile = utils.MIXED_NET_PROFILE
```

Comment:
This logic seems to be repeated in three places. Is it possible to move it into a single place?

Reply:
Have moved it to a new function.